lib/jxl/dec_frame.cc

```diff
@@ -454,7 +454,7 @@ Status FrameDecoder::ProcessDCGroup(size_t dc_group_id, BitReader* br) {
     FillImage(kInvSigmaNum / lf.epf_sigma_for_modular,
               &dec_state_->filter_weights.sigma);
   }
-  decoded_dc_groups_[dc_group_id] = true;
+  decoded_dc_groups_[dc_group_id] = 1;
```

Comment:
Can we maybe do `int{true}`?

Reply:
Any reason that's not a std::vector<bool> if that's how it is semantically used anyway?

Reply:
I assume you meant vector of bool?
That would be because vector of bool is evil (it's a template specialization that doesn't behave the way a normal vector would, which can cause all sort of surprises and has IIRC been deprecated)

Reply:
Yes, I meant vector of bool. I didn't know it was deprecated, I was hoping it would have nice specializations for things like min_element but I guess I am too optimistic. Nevermind then. Why `int{true}` and not `static_cast<uint8_t>(true)`? We are using `1` in other places already, should they all be changed? (or the `== 1` removed?)

Reply:
if you're referring to the comment below, then yes, let's remove the `== 1`.

As for `int{true}`, it guarantees at compile time that there's no conversion loss, which static_cast doesn't (although it should be `uint8_t{true}`, of course)

Reply:
Done. Also removed some similar `=! 1`.